app/models/miq_alert.rb

```diff
@@ -89,6 +103,12 @@ def validate_automate_expressions
     valid
   end
 
+  def validate_single_expression
+    unless miq_expression.nil? || hash_expression.nil?
```

Comment:
Prefer `if miq_expression && hash_expression`, which to me is a lot clearer in that it is saying "if both things are set", as opposed to "unless either thing is not set" (double-negatives and all that)

Reply:
changed :)